internal/waitext/doc.go

```diff
@@ -0,0 +1,6 @@
+package waitext
```

Comment:
Hey I think `internal` dir is not needed. 
Internal folder helps forbidding usage of packages that are meant to be used within the local codebase. I think it is most useful for protecting clients from unwanted API dependency.
Since cluster-controller is never a dependency (nobody will ever do `go mod add github.com/castai/cluster-controller`) I think we don't need `internal` folder here. What do you think?

Reply:
Sure, do you think we should move it under root (`cluster-controller/waitext/*`) or use `pkg` (`cluster-controller/pkg/waittext/*`). Second is probably more idiomatic but we don't do it for other packages in this repo. 

No preference from my side TBH

Reply:
I think under root because other packages are like that.